src/resolve/sourceComponent.ts

```diff
@@ -131,8 +133,23 @@ export class SourceComponent implements MetadataComponent {
     return this.markedForDelete;
   }
 
-  public setMarkedForDelete(asDeletion: boolean): void {
-    this.markedForDelete = asDeletion;
+  public getDestructiveChangesType(): DestructiveChangesType {
+    return this.destructiveChangesType;
+  }
+
+  public setMarkedForDelete(destructiveChangeType?: DestructiveChangesType | boolean): void {
```

Comment:
as a setter, we need to be able to specify if it's a deletion (true/false) and what type of deletion`DestructiveChangesType` passing nothing will set it to `true` and a `post` delete